server/services/permissionsService.test.ts

```diff
@@ -0,0 +1,785 @@
+
+          expect(permissions).toEqual({
+            accessCode: HmppsStatusCode.RESTRICTED_PATIENT,
+          })
+        })
+
+        it('should return OK if pom user and inactive booking user', async () => {
+          const user = { ...prisonUserMock, userRoles: [Role.InactiveBookings] }
+          const prisoner = { ...PrisonerMockDataA, restrictedPatient: true, prisonId: 'MDI' }
+
+          const permissions = await service.getOverviewPermissions(user, prisoner)
+
+          expect(permissions.accessCode).toEqual(HmppsStatusCode.OK)
+        })
+      })
+
+      describe('when prisoner is out', () => {
+        it('should return PRISONER_IS_RELEASED code if prisoner is out and user is not inactiveBookingsUser', async () => {
+          const user = prisonUserMock
+          const prisoner = { ...PrisonerMockDataA, prisonId: 'OUT' }
+
+          const permissions = await service.getOverviewPermissions(user, prisoner)
+
+          expect(permissions).toEqual({
+            accessCode: HmppsStatusCode.PRISONER_IS_RELEASED,
+          })
+        })
+
+        it('should return OK code if prisoner is out and user is inactiveBookingsUser', async () => {
+          const user = { ...prisonUserMock, userRoles: [Role.InactiveBookings] }
+          const prisoner = { ...PrisonerMockDataA, prisonId: 'OUT' }
+
+          const permissions = await service.getOverviewPermissions(user, prisoner)
+
+          expect(permissions.accessCode).toEqual(HmppsStatusCode.OK)
+        })
+
+        it('should return PRISONER_IS_TRANSFERRING code if prisoner is trn and user is not global search or inactive bookings user', async () => {
+          const user = prisonUserMock
+          const prisoner = { ...PrisonerMockDataA, prisonId: 'TRN' }
+
+          const permissions = await service.getOverviewPermissions(user, prisoner)
+
+          expect(permissions).toEqual({
+            accessCode: HmppsStatusCode.PRISONER_IS_TRANSFERRING,
+          })
+        })
+
+        it('should return OK code if prisoner is trn and user is global search user', async () => {
+          const user = { ...prisonUserMock, userRoles: [Role.GlobalSearch] }
+          const prisoner = { ...PrisonerMockDataA, prisonId: 'TRN' }
+
+          const permissions = await service.getOverviewPermissions(user, prisoner)
+
+          expect(permissions.accessCode).toEqual(HmppsStatusCode.OK)
+        })
+
+        it('should return OK code if prisoner is trn and user is inactive bookings user', async () => {
+          const user = { ...prisonUserMock, userRoles: [Role.InactiveBookings] }
+          const prisoner = { ...PrisonerMockDataA, prisonId: 'TRN' }
+
+          const permissions = await service.getOverviewPermissions(user, prisoner)
+
+          expect(permissions.accessCode).toEqual(HmppsStatusCode.OK)
+        })
+      })
+    })
+
+    describe('courtCases', () => {
+      it('should return view true if user has ReleaseDatesCalculator role', async () => {
```

Comment:
Do we want to test that it will return false otherwise? I guess I could go and just change the code to:

```
courtCases: {
  view: true,
  edit: true,
}
```

and these tests would still pass